includes/data-stores/class-wcs-related-order-store-cached-cpt.php

```diff
@@ -73,165 +75,173 @@ protected function init() {
+	 * @param int    $order_id        An order to link with the subscription.
+	 * @param int    $subscription_id A subscription to link the order to.
+	 * @param string $relation_type   The relationship between the subscription and the order. Must be 'renewal', 'switch' or 'resubscribe.
 	 */
 	protected function add_related_order_id_to_cache( $order_id, $subscription_id, $relation_type ) {
+		$subscription = $this->get_subscription_with_cache_meta( $subscription_id );
 
-		$subscription = wcs_get_subscription( $subscription_id );
-
-		// If we can't get a valid subscription, we can't update its cache
+		// If we can't get a valid subscription, we can't update its cache.
 		if ( false === $subscription ) {
 			return;
 		}
 
 		$related_order_ids = $this->get_related_order_ids( $subscription, $relation_type );
 
-		if ( ! in_array( $order_id, $related_order_ids ) ) {
-			// Add the new order to the beginning of the array to preserve sort order from newest to oldest
+		if ( ! in_array( $order_id, $related_order_ids, true ) ) {
+			// Add the new order to the beginning of the array to preserve sort order from newest to oldest.
 			array_unshift( $related_order_ids, $order_id );
-			$this->update_related_order_id_cache( $subscription_id, $related_order_ids, $relation_type );
+			$this->update_related_order_id_cache( $subscription, $related_order_ids, $relation_type );
 		}
 	}
 
 	/**
-	 * Delete a related order ID from the cached related order IDs for a given order relationship.
+	 * Deletes a related order ID from a subscription's related orders cache for a given order relationship.
 	 *
-	 * @param int $order_id The order that may be linked with subscriptions.
-	 * @param int $subscription_id A subscription to remove a linked order from.
-	 * @param string $relation_type The relationship between the subscription and the orders. Must be 'renewal', 'switch' or 'resubscribe.
+	 * @param int    $order_id        The order that may be linked with subscriptions.
+	 * @param int    $subscription_id A subscription to remove a linked order from.
+	 * @param string $relation_type   The relationship between the subscription and the orders. Must be 'renewal', 'switch' or 'resubscribe.e.
 	 */
 	protected function delete_related_order_id_from_cache( $order_id, $subscription_id, $relation_type ) {
+		$subscription = $this->get_subscription_with_cache_meta( $subscription_id );
 
-		$subscription = wcs_get_subscription( $subscription_id );
-
-		// If we can't get a valid subscription, we can't udpate its cache
+		// If we can't get a valid subscription, we can't update its cache.
 		if ( false === $subscription ) {
 			return;
 		}
 
-		$related_order_ids = $this->get_related_order_ids( $subscription, $relation_type );
+		$related_order_ids   = $this->get_related_order_ids( $subscription, $relation_type );
+		$related_order_index = array_search( $order_id, $related_order_ids, true );
```

Comment:
I just wanted to check something. I can see the strict param was added here (i.e. 3rd param in `array_search`).

Are we certain that both `$order_id` and the elements in `$related_order_ids` will be the same type? i.e. both strings or both integers?

The function docblock says that `$order_id` will be an int. Are we certain the elements in the `$related_order_ids` array will be ints?

Reply:
Yeah I confirmed this locally and included a note in the commit message. https://github.com/Automattic/woocommerce-subscriptions-core/pull/256/commits/55fb511b551e873622e8bd4ac758981d7ad310b3

It might be worth intval'ing each ID just in case.